test/Test/Data/Array/NonEmpty.purs

```diff
@@ -106,6 +114,15 @@ testNonEmptyArray = do
   assert $ NEA.elemLastIndex 1 (fromArray [1, 2, 1]) == Just 2
   assert $ NEA.elemLastIndex 4 (fromArray [1, 2, 1]) == Nothing
 
+  log "find should return the first element for which a predicate returns true in an array"
+  assert $ NEA.find (_ /= 1) (fromArray [1, 2, 1]) == Just 2
+  assert $ NEA.find (_ == 3) (fromArray [1, 2, 1]) == Nothing
```

Comment:
Should there be another test here to verify that the first element is returned when two or more elements would satisfy the predicate?

Reply:
I think so. We could also ensure that findIndex actually returns the index of the first match by changing the `_ /= 1` predicate to `_ == 1`  .

Reply:
Fixed